### DISCUSSION TITLE
Mindmapitize text terminal session

### DISCUSSION THREAD

**Discussion by metacoma** [freeplane/freeplane#1419] [Show and tell]
https://github.com/freeplane/freeplane/assets/5146707/d570e784-41a3-4bfe-adea-c538d64d3d4c

1080p: https://www.youtube.com/watch?v=znMYZOGQ3S0

**Comment by sifran-github** [comment]
impressive.
I like how it recognizes different sessions and sub sessions.

I wonder what is the purpose of it? what process it can improve?
for logging? as a good way to log your terminal work.
maybe it can be an interactive tool, and not just a log? (for example, running commands from the freeplane interface).
maybe other things?

looks nice.

**Reply by metacoma** [reply]
Hey @sifran-github thank you!

>I wonder what is the purpose of it

TL;DR: In my imagination, it's a UI/UX framework for collaborative work

> for logging? as a good way to log your terminal work. for logging? as a good way to log your terminal work.

Yes, for logging too. In my everyday work, I use a mind map to create the structure of my workday, for note-taking, and so on

> maybe it can be an interactive tool, and not just a log? (for example, running commands from the freeplane interface).

Definitely, just checkout these videos
https://www.youtube.com/watch?v=ainUEAYCHIk
https://www.youtube.com/watch?v=yBu0M8iCtVw

**Comment by nnako** [comment]
Hi @metacoma , looks impressive. Definitely. But as @sifran-github said, I can not really imagine a usecase except from providing an alternative logging / protocol means compared to the classic console log. Would you mind adding an audio track and exaplain what and why you do things? This could help imagining and grasping the ideas. But anyway, great stuff!!! ;-)

**Reply by metacoma** [reply]
hey @nnako 
thank you! 

> Would you mind adding an audio track and exaplain what and why you do things?

TL;DR: Yes, I do plan to create such a video, but for now, I'm just showing intermediate results.

My daily computer activity involves working with complex abstractions through a text terminal. I input commands, receive results, mentally interpret them, and create a mental picture of object states and their interactions.

To ease this daily routine, I started using MindMaps. The first challenge I encountered was transferring the state from my text terminal into the MindMap, which consumed time and reduced my productivity.

To manage the Mind Map's state, I developed a plugin that allows controlling the Mind Map state using third-party programs and algorithms.

I hope the video below will help you better understand my motivation and idea:


https://github.com/freeplane/freeplane/assets/5146707/f3123a56-001a-4072-b44e-bd1b09e0e31e
